src/plugins/lua/luaplugin.cpp

```diff
@@ -81,38 +107,72 @@ bool LuaPlugin::write(const Map *map, const QString &fileName)
     return true;
 }
 
-QString LuaPlugin::nameFilter() const
+QString LuaMapFormat::nameFilter() const
 {
     return tr("Lua files (*.lua)");
 }
 
-QString LuaPlugin::shortName() const
+QString LuaMapFormat::shortName() const
 {
     return QLatin1String("lua");
 }
 
-QString LuaPlugin::errorString() const
+QString LuaMapFormat::errorString() const
 {
     return mError;
 }
 
-static void writeColor(LuaTableWriter &writer,
-                       const char *name,
-                       const QColor &color)
+bool LuaTilesetFormat::supportsFile(const QString &fileName) const
 {
-    // Example: backgroundcolor = { 255, 200, 100 }
-    writer.writeStartTable(name);
-    writer.setSuppressNewlines(true);
-    writer.writeValue(color.red());
-    writer.writeValue(color.green());
-    writer.writeValue(color.blue());
-    if (color.alpha() != 255)
-        writer.writeValue(color.alpha());
-    writer.writeEndTable();
-    writer.setSuppressNewlines(false);
+    return true;
+}
+
+bool LuaTilesetFormat::writeTileset(const Tileset &tileset, const QString &fileName)
+{
+  SaveFile file(fileName);
```

Comment:
Coding style: please correct the indentation.

Reply:
Done